provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -276,7 +276,10 @@ func generateSchema() schema.PackageSpec {
 							"Also consider setting `nodeAssociatePublicIpAddress: false` for fully private workers.",
 					},
 					"nodeGroupOptions": {
-						TypeSpec:    schema.TypeSpec{Ref: "#/types/eks:index:ClusterNodeGroupOptions"},
+						TypeSpec: schema.TypeSpec{
+							Ref:   "#/types/eks:index:ClusterNodeGroupOptions",
+							Plain: true,
+						},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/cluster.ts#L958-L961